src/llm/backends/openai.py

```
@@ -241,14 +241,23 @@ async def stream(
+            # Inject schema into the last user message so the model knows the
+            # expected output format — mirrors what the Anthropic path does.
+            schema_json = json.dumps(response_format.model_json_schema(), indent=2)
+            msgs = params.get("messages", [])
+            if msgs:
+                last_msg = msgs[-1]
+                if isinstance(last_msg.get("content"), str):
+                    last_msg = dict(last_msg)
+                    last_msg["content"] += (
+                        f"\n\nRespond with valid JSON matching this schema:\n{schema_json}"
+                    )
+                    params["messages"] = list(msgs[:-1]) + [last_msg]
```

Ruff (0.15.11)

[warning] Consider `[*list(msgs[:-1]), last_msg]` instead of concatenation

Replace with `[*list(msgs[:-1]), last_msg]`

(RUF005)